databricks/sdk/core.py

```diff
@@ -292,6 +295,37 @@ def inner() -> Dict[str, str]:
     return inner
 
 
+def get_token(cfg: 'Config', resource: str) -> AzureCliTokenSource:
```

Comment:
why can't this validation be moved to `AzureCliTokenSource` constructor?

Reply:
That would mean adding a try/except block around the "super().__init__" code. Not being that familiar with Python, and coming from languages like Java which don't allow you to do that, it seemed like a bad idea to me. But I can add it if it is considered acceptable practice in Python.

Reply:
can you then try adding `validate` (or validate_subscription, or something like that)  method on `AzureCliTokenSource` and call it instead of the first call to `token`. 

```
    try:
        token_source.token()
    except FileNotFoundError:
```

Reply:
btw, the `IncorrectClaimException` in the original ticket does refer to Azure **Tenant** (identities), not the Azure **Subscription** (resources). there are valid cases, when user has no access to subscription, but has access to a workspace. also, `azure_workspace_resource_id` is not required for Azure CLI auth type.

to solve `IncorrectClaimException` we have to supply `azure_login_app_id`. I'll reach to you on slack.